src/client/opamAction.ml

```diff
@@ -521,31 +521,81 @@ let prepare_package_source st nv dir =
 
 let compilation_env t opam =
   let open OpamParserTypes in
+  let empty = Some (SPF_Resolved None) in
   let build_env =
-    List.map (OpamEnv.env_expansion ~opam t) (OpamFile.OPAM.build_env opam)
+    List.map
+      (fun env ->
+         OpamEnv.resolve_separator_and_format
+           (OpamEnv.env_expansion ~opam t env))
+      (OpamFile.OPAM.build_env opam)
   in
   let cygwin_env =
     match OpamSysInteract.Cygwin.cygbin_opt t.switch_global.config with
     | Some cygbin ->
-      [ "PATH", EqPlus, OpamFilename.Dir.to_string cygbin, Some "Cygwin path" ]
+      [{ envu_var = "PATH";
+         envu_op = EqPlus;
+         envu_value = OpamFilename.Dir.to_string cygbin;
+         envu_comment = Some "Cygwin path";
+         envu_rewrite = empty;
+       }]
```

Comment:
Added in a commit

Reply:
I don't know which version is more easily readable

Reply:
Sorry - looking at it again, I now realise that I'd forgotten that a GADT was involved and that we can't convey that in optional parameters! (yet... it's been on my wish list for quite a few years...)

Wouldn't it be OK not to label those three arguments, though> I think `env_update_resolved "PATH" EqPlus (OpamFilename.Dir.to_string cygbin` quite clearly conveys what it's doing to give:

```ocaml
      [ OpamTypesBase.env_update_resolved ~comment:"Cygwin path"
          "PATH" EqPlus (OpamFilename.Dir.to_string cygbin) ]
```

?

Reply:
It is possible. The labels came at first because of var & values are string, and like that there is no confusion between them.
I'm ok to remove them and have then
```ocaml
val env_update_resolved:
  ?comment:string -> ?rewrite:spf_resolved separator_path_format option
  -> string -> env_update_op_kind -> string
  -> spf_resolved env_update
```